internal/outputgenerators/bonnmotion.go

```diff
@@ -156,27 +164,43 @@ func (b Bonnmotion) Generate(exp experiment.Experiment) {
 	}
 	b.outputFolder = outputFolder
 
-	b.stepFilePath = filepath.Join(b.outputFolder, BonnMotionStepFile)
-	allowedToWriteStepFile := folderstructure.MayCreatePath(b.stepFilePath)
+	stepFilePath := filepath.Join(b.outputFolder, BonnMotionStepFile)
+	allowedToWriteStepFile := folderstructure.MayCreatePath(stepFilePath)
 	if !allowedToWriteStepFile {
 		logger.Error("Not allowed to write step file!")
 		return
 	}
+	b.stepFile, err = os.OpenFile(stepFilePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		logger.Error("Error opening step file:", err)
+		return
+	}
+	defer func() {
+		if cerr := b.stepFile.Close(); cerr != nil {
```

Comment:
if possible please define cerr outside of the if statement for readabillity reason

Reply:
That would technically be possible, but would be inconsistent with [core.go](https://github.com/stg-tud/bp2022_netlab/blob/main/internal/outputgenerators/core.go#L257), [corremulab.go](https://github.com/stg-tud/bp2022_netlab/blob/main/internal/outputgenerators/coreemulab.go#L83) and [theone.go](https://github.com/stg-tud/bp2022_netlab/blob/main/internal/outputgenerators/theone.go#L158).